lib/TwoFactorAuth.php

```diff
@@ -174,19 +169,7 @@ public function getQrCodeProvider(): IQRCodeProvider
      */
     public function getRngProvider(): IRNGProvider
     {
-        if ($this->rngprovider !== null) {
-            return $this->rngprovider;
-        }
-        if (function_exists('random_bytes')) {
-            return $this->rngprovider = new CSRNGProvider();
-        }
-        if (function_exists('openssl_random_pseudo_bytes')) {
-            return $this->rngprovider = new OpenSSLRNGProvider();
-        }
-        if (function_exists('hash')) {
-            return $this->rngprovider = new HashRNGProvider();
-        }
-        throw new TwoFactorAuthException('Unable to find a suited RNGProvider');
+        return $this->rngprovider ?? new CSRNGProvider();
```

Comment:
this line doesn't set `$this->rngprovider` any more so should probably do that

Reply:
You're correct that the behavior is changed in the sense that it doesn't populate this value, but given that:

* this attribute is private
* it's only use here

Then it's pointless to populate this class property. Unless I'm missing something.

Reply:
if you swapped it to `??=` then it would match the other two providers so they all behave in the same way

Reply:
@willpower232 Yeah, it doesn't hurt, and aligns with the other providers, so I've pushed a change with your suggestion :)